src/screens/WalletConnect/WalletConnectList.js

```diff
@@ -0,0 +1,178 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { t } from 'ttag';
+import { get } from 'lodash';
+import {
+  StyleSheet,
+  SafeAreaView,
+  Text,
+  View,
+  Image,
+  TouchableOpacity,
+  Alert,
+} from 'react-native';
+import FeedbackModal from '../../components/FeedbackModal';
+import errorIcon from '../../assets/images/icErrorBig.png';
+
+import HathorHeader from '../../components/HathorHeader';
+import SimpleButton from '../../components/SimpleButton';
+import { HathorList } from '../../components/HathorList';
+import {
+  walletConnectCancelSession,
+  setWCConnectionFailed,
+} from '../../actions';
+
+const connectFailedText = t`There was an error connecting. Please try again later.`;
+
+const style = StyleSheet.create({
+  componentWrapper: {
+    flex: 1,
+    justifyContent: 'center',
+    alignItems: 'center',
+    backgroundColor: '#f7f7f7',
+    alignSelf: 'stretch',
+  },
+  safeAreaView: {
+    flex: 1,
+    justifyContent: 'center',
+    alignItems: 'center',
+    alignSelf: 'stretch',
+  },
+  buttonWrapper: {
+    flexDirection: 'row',
+    alignItems: 'center',
+    borderBottomWidth: 1,
+    borderBottomColor: '#cecece',
+    paddingBottom: 16,
+  },
+  image: {
+    width: 50,
+    height: 50,
+    borderRadius: 25,
+    marginTop: 16,
+    marginLeft: 16,
+  },
+  dataWrapper: {
+    flexDirection: 'column',
+    marginLeft: 16,
+    marginRight: 16,
+    marginTop: 16,
+  },
+  sessionName: {
+    marginBottom: 8,
+    color: 'rgba(0, 0, 0, 0.87)',
+    fontSize: 12,
+    fontWeight: 'bold',
+  },
+  sessionData: {
+    marginBottom: 8,
+    color: 'rgba(0, 0, 0, 0.57)',
+    fontSize: 12,
+  },
+  sessionListWrapper: {
+    flex: 1,
+    justifyContent: 'center',
+    alignItems: 'center',
+    margin: 8,
+    alignSelf: 'stretch',
+    borderRadius: 30,
+  }
+});
+
+export default function WalletConnectList({ navigation }) {
+  const dispatch = useDispatch();
+  const connectionFailed = useSelector((state) => state.walletConnect.connectionFailed);
```

Comment:
After adding a new connection we go back to this page, so it will check if the connection failed or not here

The connection might fail before or after I change page, it shouldn't matter, we will show the modal after the redux is set